src/libs/actions/Card.js

```diff
@@ -101,4 +102,30 @@ function clearCardListErrors(cardID) {
     Onyx.merge(ONYXKEYS.CARD_LIST, {[cardID]: {errors: null, isLoading: false}});
 }
 
-export {reportVirtualExpensifyCardFraud, activatePhysicalExpensifyCard, clearCardListErrors};
+/**
+ * Makes an API call to get virtual card details (pan, cvv, expiration date, address)
+ * This function purposefully uses `makeRequestWithSideEffects` method. For security reason
+ * card details cannot be persisted in Onyx and have to be asked for each time a user want's to
+ * reveal them.
+ *
+ * @param {String} cardID - virtual card ID
+ *
+ * @returns {Promise<Object>} - promise with card details object
+ */
+function revealVirtualCardDetails(cardID) {
+    return new Promise((resolve, reject) => {
```

Comment:
Why are we wrapping the call in this Promise? We don't use this pattern anywhere else

Reply:
@lukemorawski if you could explain, just curious

Reply:
I think something like this would also work..

```js
function revealVirtualCardDetails(cardID) {
    // eslint-disable-next-line rulesdir/no-api-side-effects-method
    return API.makeRequestWithSideEffects('RevealVirtualCardDetails', {cardID})
        .then((response) => {
            if (response.jsonCode !== CONST.JSON_CODE.SUCCESS) {
                throw Localize.translateLocal('cardPage.cardDetailsLoadingFailure');
            }

            return response;
        })
        .catch((err) => {
            throw err.message;
        });
}
```

Reply:
cc @lukemorawski @rushatgabhane